servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletHttpRequest.java

```diff
@@ -0,0 +1,624 @@
+
+    @Override
+    public void setCharacterEncoding(String env) throws UnsupportedEncodingException {
+        throw new IllegalStateException("Can't set character encoding after request is initialized");
+    }
+
+    @Override
+    public int getContentLength() {
+        return httpRequest.content().length();
+    }
+
+    @Override
+    public long getContentLengthLong() {
+        return getContentLength();
+    }
+
+    @Override
+    @Nullable
+    public String getContentType() {
+        final MediaType contentType = httpRequest.headers().contentType();
+        return contentType == null ? null : contentType.toString();
+    }
+
+    @Override
+    public DefaultServletInputStream getInputStream() throws IOException {
+        if (reader != null) {
+            throw new IllegalStateException("getReader() has already been called for this request");
+        }
+        useInputStream = true;
+        return inputStream;
+    }
+
+    @Override
+    @Nullable
+    public String getParameter(String name) {
+        requireNonNull(name, "name");
+        return queryParams.get(name);
+    }
+
+    @Override
+    public Enumeration<String> getParameterNames() {
+        return Collections.enumeration(ImmutableSet.copyOf(queryParams.names()));
+    }
+
+    @Override
+    @Nullable
+    public String[] getParameterValues(String name) {
+        requireNonNull(name, "name");
+        if (queryParams.getAll(name) == null) {
+            return null;
+        }
+        return queryParams.getAll(name).toArray(new String[0]);
+    }
+
+    @Override
+    public Map<String, String[]> getParameterMap() {
+        return parameters;
+    }
+
+    @Override
+    public String getProtocol() {
+        return serviceRequestContext.sessionProtocol().uriText();
+    }
+
+    @Override
+    @Nullable
+    public String getScheme() {
+        return httpRequest.scheme();
+    }
+
+    @Override
+    public String getServerName() {
+        return serviceRequestContext.config().virtualHost().defaultHostname();
+    }
+
+    @Override
+    public int getServerPort() {
+        return serviceRequestContext.config().server().activeLocalPort();
+    }
+
+    @Override
+    public BufferedReader getReader() throws IOException {
+        if (reader == null && !useInputStream) {
+            synchronized (this) {
+                if (reader == null) {
+                    reader = new BufferedReader(
+                            new InputStreamReader(inputStream, getCharacterEncoding()));
+                }
+            }
+        }
+        return reader;
+    }
+
+    @Override
+    public String getRemoteAddr() {
+        return ((InetSocketAddress) serviceRequestContext.remoteAddress()).getAddress().getHostAddress();
+    }
+
+    @Override
+    public String getRemoteHost() {
+        return ((InetSocketAddress) serviceRequestContext.remoteAddress()).getHostName();
+    }
+
+    @Override
+    public int getRemotePort() {
+        return getServerPort();
```

Comment:
We should return the remote port:
`((InetSocketAddress) serviceRequestContext.remoteAddress()).getPort()`